### PR TITLE
add Pkg.precompile

### PR DESCRIPTION
**Pull request by sivapvarma**
some discussion at [julia discourse](https://discourse.julialang.org/t/precompile-only-available-in-repl-mode/24115)

### CODE REVIEW

File: src/Pkg.jl [hunk 1]
```diff
@@ -97,6 +97,18 @@ See also [`PackageSpec`](@ref).
...
 const add = API.add
 
+"""
+    Pkg.precompile()
+
+Precompile all the dependencies of the project by running `import` on all of them in a new process.
+
+# Examples
+```julia
+Pkg.precompile() # precompiles all the packages in this project.
```
Comment (fredrikekre) [inline src/Pkg.jl]: Don't think this is needed since it is included in just above:
```suggestion
Pkg.precompile()
```

**Comment by fredrikekre** [comment]
Can you also add `Pkg.precompile` to the list here: https://raw.githubusercontent.com/JuliaLang/Pkg.jl/master/docs/src/api.md for example just after `Pkg.status`?

**Comment by sivapvarma** [comment]
added `Pkg.precompile` in docs as well.

File: src/Pkg.jl [hunk 2]
```diff
@@ -97,6 +97,18 @@ See also [`PackageSpec`](@ref).
 """
 const add = API.add
 
+"""
+    Pkg.precompile()
+
+Precompile all the dependencies of the project by running `import` on all of them in a new process.
```
Comment (KristofferC) [inline src/Pkg.jl]:
```suggestion
Precompile all the dependencies of the project.
```

It does not run `import` on them in a new process anymore.

Comment (sivapvarma) [reply src/Pkg.jl]: Done.

**Comment by fredrikekre** [comment]
Derp. Why do GitHub close a PR when beeing force pushed to?

**Pull request closed without merging**